speech/grpc/transcribe_streaming.py

```diff
@@ -86,54 +84,48 @@ def _audio_data_generator(buff):
             except queue.Empty:
                 break
 
-        # If `_fill_buffer` adds `None` to the buffer, the audio stream is
-        # closed. Yield the final bit of the buffer and exit the loop.
+        # `None` in the buffer signals that the audio stream is closed. Yield
+        # the final bit of the buffer and exit the loop.
         if None in data:
             stop = True
             data.remove(None)
         yield b''.join(data)
 
 
-def _fill_buffer(audio_stream, buff, chunk, stoprequest):
+def _fill_buffer(buff):
```

Comment:
Instead of a closure, just use `functools.partial`:

```python
def audio_stream_callback(buffer, in_data, frame_count, time_info, status_flags):
    ...

stream_callback=functools.partial(audio_stream_callback, buffer)
```